### ISSUE TITLE
Question: Is there a callback function that fires when the audio segment becomes larger than minSpeechFrames?

### ISSUE THREAD

**Issue by jin60641** [ricky0123/vad#67]
when misfire (audio segment is smaller than minSpeechFrames)
1. onSpeechStart
2. onVADMisfire

when fire (audio segment is bigger than minSpeechFrames)
1. onSpeechStart
2. **onSpeechRealStart** (This callback function fires when the audio segment becomes larger than minSpeechFrames)
3. onSpeechEnd

I need the **onSpeechRealStart**(Provisional name) function, but I can't find it in the documentation. If it exists, please explain it. Thanks!

**Comment by HayatoYagi** [comment]
Currently, there is no such function🥲
Could you please explain in more detail why you want that?

**Comment by jin60641** [comment]
Thanks for reply
I'd like to perform a high-cost computation with the audio segment onSpeechEnd, and if the user creates a new audio segment, I want to immediately cancel the computation and perform it again with the new audio segment. If there was an onSpeechRealStart event, I could cancel the computation even sooner and save on cost.

**Comment by HayatoYagi** [comment]
Thanks. It sounds reasonable to me to add `onSpeechRealStart`.
Another solution is that you can `setTimeout` in `onSpeechStart` with a bit longer time than `minSpeechFrames` and `clearTimeout` in `onVadMisfire` (Is this whay you are doing now?)

@ricky0123 What do you think?

**Comment by jin60641** [comment]
@HayatoYagi That's right. As I wrote in the issue, I used setTimeout to check if the userSpeaking flag remains true, and solved the problem circumventively. However, since this doesn't exactly match minSpeechFrames, I created the issue.

**Comment by ricky0123** [comment]
Hi @jin60641, thanks for your comments, this is a very interesting request that I had never thought of. I can see the utility of an onSpeechRealStart callback. Just to clarify, though, minSpeechFrames refers to the minimum number of frames that should have speech probability greater than positiveSpeechThreshold. So, onSpeechRealStart won't always fire minSpeechFrames after the first frame with speech is detected. We need to continuously tally the number of frames with speech probability greater than positiveSpeechThreshold and fire onSpeechRealStart after that tally exceeds minSpeechFrames. I just wanted to clarify that in case anyone starts implementing it.

**Comment by jin60641** [comment]
@ricky0123 
Thanks!
Your explanation matches my intention for onSpeechRealStart. It should be called not only when the first frame comes in, but also when the number of frames exceeding the positiveSpeechThreshold has passed minSpeechFrames, ensuring that it will not be a misfire.

**Comment by Oudwins** [comment]
This feature has been pushed to the docs but it doesn't appear to have landed on the npm package.

**Comment by ricky0123** [comment]
@Oudwins yes, I have to fix something before releasing the next batch of changes. Will get to it in the next week or so

**Comment by swift4ever** [comment]
gentlemen, do we have any update regarding landing this on the npm package?

**Comment by ricky0123** [comment]
Hi all, it's been released in vad-web 0.0.23 and vad-react 0.0.29
